### PR TITLE
[TM-874] site profile non monitoring partner

### PR DESCRIPTION
**Pull request by egrojMonroy**
https://gfw.atlassian.net/browse/TM-874

### CODE REVIEW

File: src/components/elements/Map-mapbox/hooks/useMap.ts [hunk 1]
```diff
@@ -46,6 +46,8 @@ export const useMap = (onSave?: (geojson: any, record: any) => void) => {
 
   const initMap = () => {
     if (map.current) return;
+
+    console.log("map.current", map.current);
```
Comment (roguenet) [inline src/components/elements/Map-mapbox/hooks/useMap.ts:50]: 🛅

File: src/components/elements/MapPolygonPanel/AttributeInformation.tsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  {
+    title: "Full",
+    value: 3
+  }
+];
+const AttributeInformation = () => {
+  return (
+    <div className="flex flex-col gap-4">
+      <Input
+        label="Polygon Name"
```
Comment (roguenet) [inline src/components/elements/MapPolygonPanel/AttributeInformation.tsx:73]: Please make sure all free text in this component is wrapped in `t()`

File: src/components/elements/MapPolygonPanel/ChecklistInformation.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+import Button from "../Button/Button";
+import Text from "../Text/Text";
+
+const ChecklistInformation = () => {
+  const { openModal, closeModal } = useModalContext();
+  const openFormModalHandlerRequestPolygonSupport = () => {
+    openModal(
+      <ModalWithMap
+        title="Request Support"
```
Comment (roguenet) [inline src/components/elements/MapPolygonPanel/ChecklistInformation.tsx:13]: Looks like this component need `t()` as well.

File: src/components/elements/MapPolygonPanel/ChecklistInformation.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        onClose={closeModal}
+        content="Faja Lobi Project&nbsp;&nbsp;•&nbsp;&nbsp;Priceless Planet Coalition"
```
Comment (roguenet) [inline src/components/elements/MapPolygonPanel/ChecklistInformation.tsx:15]: Please remove all placeholder / mocked data before merging to staging.

File: src/components/elements/MapPolygonPanel/MapEditPolygonPanel.tsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+}: MapEditPolygonPanelProps) => {
+  return (
+    <>
+      <div className="flex items-start justify-between gap-4">
+        <div>
+          <Text variant="text-12-light" className="text-white ">
+            Faja Lobi Project
+          </Text>
+          <Text variant="text-20-bold" className="mb-4 text-white">
+            Iseme Site
```
Comment (roguenet) [inline src/components/elements/MapPolygonPanel/MapEditPolygonPanel.tsx:35]: Please remove all mocked data before merging to staging

File: src/components/elements/MapPolygonPanel/MapEditPolygonPanel.tsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+  setPreviewVersion: Dispatch<SetStateAction<boolean>>;
+}
+
+const MapEditPolygonPanel = ({
+  setEditPolygon,
+  tabEditPolygon,
+  setTabEditPolygon,
+  setPreviewVersion
+}: MapEditPolygonPanelProps) => {
+  return (
```
Comment (roguenet) [inline src/components/elements/MapPolygonPanel/MapEditPolygonPanel.tsx:27]: Avoid return statement for single-statement arrow functions

File: src/components/elements/MapPolygonPanel/MapPolygonCheckPanelItem.tsx [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+  polygon,
+  status,
+  ...props
+}: MapPolygonCheckPanelItemProps) => {
+  const { openModal, closeModal } = useModalContext();
+  const [openCollapse, setOpenCollapse] = useState(true);
+  const openFormModalHandlerRequestPolygonSupport = () => {
+    openModal(
+      <ModalWithMap
+        title="Request Support"
```
Comment (roguenet) [inline src/components/elements/MapPolygonPanel/MapPolygonCheckPanelItem.tsx:43]: Please wrap all free text in this component in `t()`

File: src/components/elements/MapPolygonPanel/MapPolygonCkeckPanel.tsx [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+const MapPlygonCheckPanel = ({ emptyText, onLoadMore, setEditPolygon, selected }: MapPolygonCheckPanelProps) => {
+  const t = useT();
+
+  const refContainer = useRef<HTMLDivElement>(null);
+  return (
+    <>
+      <Text variant="text-14" className="mb-6 text-white">
+        Available polygons
```
Comment (roguenet) [inline src/components/elements/MapPolygonPanel/MapPolygonCkeckPanel.tsx:26]: Please wrap all free text in `t()`

File: src/components/elements/MapPolygonPanel/MapPolygonCkeckPanel.tsx [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+      </Text>
+      <div className="h-[calc(100%-150px)] rounded-bl-lg">
+        {PolygonAvailableData.length === 0 && (
+          <Text variant="text-16-light" className="mt-8 text-white">
+            {emptyText || t("No result")}
```
Comment (roguenet) [inline src/components/elements/MapPolygonPanel/MapPolygonCkeckPanel.tsx:31]: `??`

File: src/components/elements/MapPolygonPanel/MapPolygonPanelItem.tsx [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+  setEditPolygon,
+  ...props
+}: MapPolygonPanelItemProps) => {
+  const { openModal, closeModal } = useModalContext();
+  const openFormModalHandlerRequestPolygonSupport = () => {
+    openModal(
+      <ModalWithMap
+        title="Request Support"
+        onClose={closeModal}
+        content="Faja Lobi Project&nbsp;&nbsp;•&nbsp;&nbsp;Priceless Planet Coalition"
```
Comment (roguenet) [inline src/components/elements/MapPolygonPanel/MapPolygonPanelItem.tsx:40]: This component also needs `t()` on text and mocked data removed.

File: src/components/elements/MapPolygonPanel/MapPolygonSitePanel.tsx [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+        if (ref) {
+          ref.removeEventListener("change", handleChange);
+        }
+      });
+    };
+  }, [openMenu]);
+
+  return (
+    <>
+      <FilterSearchBox placeholder={"Search"} className="mb-4 w-full" onChange={() => {}} />
```
Comment (roguenet) [inline src/components/elements/MapPolygonPanel/MapPolygonSitePanel.tsx:78]: `t()`

File: src/components/elements/MapPolygonPanel/MapPolygonSitePanel.tsx [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+          <div className="rounded bg-white p-1.5">
+            <Icon name={IconNames.IC_SORT} className="h-4 w-4 text-blueCustom-900 hover:text-primary-500" />
+          </div>
+        </div>
+      </div>
+
+      <div className="h-[calc(100%-159px)] rounded-bl-lg">
+        {PolygonData.length === 0 && (
+          <Text variant="text-16-light" className="mt-8 text-white">
+            {emptyText || t("No result")}
```
Comment (roguenet) [inline src/components/elements/MapPolygonPanel/MapPolygonSitePanel.tsx:150]: `??`

File: src/components/elements/MapPolygonPanel/VersionInformation.tsx [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+import { MENU_PLACEMENT_RIGHT_BOTTOM } from "../Menu/MenuVariant";
+import Text from "../Text/Text";
+
+const VersionInformation = ({ setPreviewVersion }: { setPreviewVersion: Dispatch<SetStateAction<boolean>> }) => {
+  const { openModal, closeModal } = useModalContext();
+  const openFormModalHandlerConfirm = () => {
+    openModal(
+      <ModalConfirm
+        title={"Confirm Version Delete?"}
+        content="Do you want to delete this version?"
```
Comment (roguenet) [inline src/components/elements/MapPolygonPanel/VersionInformation.tsx:18]: `t()` in this component

File: src/components/elements/MapPolygonPanel/VersionInformation.tsx [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+      title: "ID Wenguru v1",
+      date: "Feb 8, 24",
+      current: "No"
+    },
+    {
+      title: "ID Wenguru v1",
+      date: "Feb 6, 24",
+      current: "No"
+    }
+  ];
```
Comment (roguenet) [inline src/components/elements/MapPolygonPanel/VersionInformation.tsx:72]: Please remove all mocked data in component code.

File: src/pages/project/[uuid]/components/OverviewMapArea.tsx [hunk 15]
```diff
@@ -4,48 +4,40 @@ import { useEffect, useState } from "react";
...
+  const setResultValues = (result: any) => {
+    if (result.polygonsData) {
+      setPolygonsData(result.polygonsData);
+      setEntityBbox(result.bbox as BBox);
+    }
   };
 
   useEffect(() => {
-    if (project?.uuid) {
+    console.log("entityModel", entityModel);
```
Comment (roguenet) [inline src/pages/project/[uuid]/components/OverviewMapArea.tsx:35]: 🛅

File: src/pages/site/[uuid]/components/SiteArea.tsx [hunk 16]
```diff
@@ -0,0 +1,85 @@
...
+            </Button>
+          </div>
+        </When>
+        <When condition={!!previewVersion}>
+          <div className="absolute bottom-8 left-[54vw] z-20 w-[22vw] rounded bg-white p-3">
+            <button className="absolute top-3 right-4 hover:opacity-60" onClick={() => setPreviewVersion(false)}>
+              <Icon name={IconNames.CLEAR} className="h-3 w-3 wide:h-4 wide:w-4" />
+            </button>
+            <Text variant="text-10-bold" className="mb-4 text-center">
+              Preview Attributes
```
Comment (roguenet) [inline src/pages/site/[uuid]/components/SiteArea.tsx:45]: `t()` throughout this component.

File: src/pages/site/[uuid]/components/SiteArea.tsx [hunk 17]
```diff
@@ -0,0 +1,85 @@
...
+            </Text>
+            <div className="grid grid-cols-2 gap-4 border-b border-grey-750 py-2">
+              <Text variant="text-10-light" className="opacity-60">
+                Polygon ID
+              </Text>
+              <Text variant="text-10-light">1213023412</Text>
```
Comment (roguenet) [inline src/pages/site/[uuid]/components/SiteArea.tsx:51]: This looks like mocked data? Please ensure there is no mocked data in the compoenntn code when it's ready to merge to staging.

File: src/components/elements/MapPolygonPanel/AttributeInformation.tsx (hunk 2 above)
Comment (dottyy) [reply src/components/elements/MapPolygonPanel/AttributeInformation.tsx]: done

File: src/components/elements/MapPolygonPanel/ChecklistInformation.tsx (hunk 3 above)
Comment (dottyy) [reply src/components/elements/MapPolygonPanel/ChecklistInformation.tsx]: done

File: src/components/elements/MapPolygonPanel/ChecklistInformation.tsx (hunk 4 above)
Comment (dottyy) [reply src/components/elements/MapPolygonPanel/ChecklistInformation.tsx]: done

Comment (dottyy) [reply src/components/elements/MapPolygonPanel/ChecklistInformation.tsx]: done

File: src/pages/site/[uuid]/components/SiteArea.tsx (hunk 16 above)
Comment (dottyy) [reply src/pages/site/[uuid]/components/SiteArea.tsx]: done

File: src/pages/site/[uuid]/components/SiteArea.tsx (hunk 17 above)
Comment (dottyy) [reply src/pages/site/[uuid]/components/SiteArea.tsx]: done

File: src/components/elements/MapPolygonPanel/VersionInformation.tsx (hunk 14 above)
Comment (dottyy) [reply src/components/elements/MapPolygonPanel/VersionInformation.tsx]: done

File: src/components/elements/MapPolygonPanel/VersionInformation.tsx (hunk 13 above)
Comment (dottyy) [reply src/components/elements/MapPolygonPanel/VersionInformation.tsx:21]: done

File: src/components/elements/MapPolygonPanel/MapPolygonSitePanel.tsx (hunk 11 above)
Comment (dottyy) [reply src/components/elements/MapPolygonPanel/MapPolygonSitePanel.tsx]: done

File: src/components/elements/MapPolygonPanel/MapPolygonSitePanel.tsx (hunk 12 above)
Comment (dottyy) [reply src/components/elements/MapPolygonPanel/MapPolygonSitePanel.tsx]: done

File: src/components/elements/MapPolygonPanel/MapPolygonPanelItem.tsx (hunk 10 above)
Comment (dottyy) [reply src/components/elements/MapPolygonPanel/MapPolygonPanelItem.tsx]: done

File: src/components/elements/MapPolygonPanel/VersionInformation.tsx [hunk 18]
```diff
@@ -37,78 +40,51 @@ const VersionInformation = ({ setPreviewVersion }: { setPreviewVersion: Dispatch
...
+          -
+        </Text>
+        <Text variant="text-10" className="text-white">
+          -
+        </Text>
+        <div className="flex justify-between">
+          <button
+            className={classNames("text-10-bold w-[64%] rounded-md border border-white", {
+              "bg-white text-[#797F62]": true, // Replace 'item.current === "Yes"' with 'true' or provide a valid variable
+              "bg-transparent text-white": false // Replace 'item.current === "No"' with 'false' or provide a valid variable
```
Comment (roguenet) [inline src/components/elements/MapPolygonPanel/VersionInformation.tsx:75]: I think these comments are 🛅 ?

File: src/components/elements/MapPolygonPanel/VersionInformation.tsx (hunk 13 above)
Comment (roguenet) [reply src/components/elements/MapPolygonPanel/VersionInformation.tsx:21]: These two still look like they need `t()`, but I see it was added in a lot of other places in the component.

**Review by roguenet — COMMENTED** [review]
I've resolved all the comments that were addressed, but there are a few left that seem to have been missed.

File: src/components/elements/MapPolygonPanel/MapEditPolygonPanel.tsx (hunk 5 above)
Comment (dottyy) [reply src/components/elements/MapPolygonPanel/MapEditPolygonPanel.tsx]: done

File: src/components/elements/MapPolygonPanel/MapPolygonCkeckPanel.tsx (hunk 8 above)
Comment (dottyy) [reply src/components/elements/MapPolygonPanel/MapPolygonCkeckPanel.tsx]: done

File: src/components/elements/MapPolygonPanel/MapPolygonCkeckPanel.tsx (hunk 9 above)
Comment (dottyy) [reply src/components/elements/MapPolygonPanel/MapPolygonCkeckPanel.tsx]: done

File: src/components/elements/MapPolygonPanel/VersionInformation.tsx (hunk 13 above)
Comment (dottyy) [reply src/components/elements/MapPolygonPanel/VersionInformation.tsx]: done

File: src/components/elements/MapPolygonPanel/VersionInformation.tsx (hunk 18 above)
Comment (dottyy) [reply src/components/elements/MapPolygonPanel/VersionInformation.tsx]: done

**Review by roguenet — APPROVED** [review]

**Pull request merged**